engine/runtime/src/main/java/org/enso/interpreter/node/EnsoRootNode.java

```diff
@@ -145,10 +143,18 @@ public LocalScope getLocalScope() {
    *
    * @return the module scope for this node
    */
-  public ModuleScope getModuleScope() {
+  public ModuleScope.Builder getModuleScope() {
```

Comment:
Ideally such a `public` getter of the builder should be avoided. The `Builder` should be `private` to the creator of it, not floating around being accessed by random parties.

Reply:
Yes, I was hoping that would be the case as well, except that we are currently dealing with a situation when the builder's state is also inspected while stuff is added to it.